OloEditor/src/Panels/SceneHierarchyPanel.cpp

```
@@ -211,30 +390,204 @@ namespace OloEngine
+            }
+            else if (shift && m_SelectionContext)
+            {
+                // Shift+click: select range between last clicked and this entity
+                // Collect visible entities in order
+                std::vector<Entity> visible;
+                m_Context->m_Registry.view<entt::entity>().each([&](const auto e)
+                                                                { visible.emplace_back(e, *m_Context); });
+
+                auto itA = std::ranges::find(visible, m_SelectionContext);
+                auto itB = std::ranges::find(visible, entity);
+                if (itA != visible.end() && itB != visible.end())
```

Clang (14.0.6)

[note] +3, including nesting penalty of 2, nesting level increased to 3

(clang)